tests/test_initrd.py

```diff
@@ -124,6 +124,48 @@ def test_extract_modules_from_deb(self):
         assert os.path.isfile(os.path.join(
             self.generator.config.target_dir, 'lib', 'modules', kversion, module))
 
+    @pytest.mark.requires_download
+    def test_extract_compressed_modules_from_deb(self):
+        """ Test compressed modules. """
+        test_dir = os.path.dirname(os.path.abspath(__file__))
+        yaml = os.path.join(test_dir, 'data', 'initrd_noble.yaml')
+        # Prepare generator
+        temp_dir = tempfile.mkdtemp()
+        generator = InitrdGenerator(yaml, temp_dir)
```

Comment:
pytest brings fixtures for temporary files, that are cleaned up later automatically, but also kept for manual inspection. Using them instead of `tempfile.mkdtemp()` makes it easier to debug

```suggestion
    def test_extract_compressed_modules_from_deb(self, tmp_path: Path):
        """ Test compressed modules. """
        test_dir = os.path.dirname(os.path.abspath(__file__))
        yaml = os.path.join(test_dir, 'data', 'initrd_noble.yaml')
        # Prepare generator
        generator = InitrdGenerator(yaml, str(tmp_path))
```